src/components/UserDashboard/CavaticaProjects/index.js

```diff
@@ -1,129 +1,35 @@
-          </CavaticaProvider>
-        ),
-      },
-      {
-        title: 'Create',
-        component: <Create setBadge={setBadge} setStackIndex={index => setStackIndex(index)} />,
-      },
-    ];
-
-    const isConnected = isValidKey(integrationTokens[CAVATICA]);
-
-    const Header = (
-      <CardHeader title="Cavatica Projects" badge={badgeNumber}>
-        {!isConnected &&
-          cardStack.map((card, i) => (
-            <DualPaneHeader
-              key={i}
-              active={i === stackIndex}
-              onClick={() => setStackIndex(i)}
-              title={card.title}
-            />
-          ))}
-      </CardHeader>
-    );
-
-    const activeCard = cardStack[stackIndex];
-
-    return (
-      <DualPaneCard
-        Header={Header}
-        stackIndex={stackIndex}
-        setStackIndex={index => setStackIndex(index)}
-        inactive={!isConnected}
-        scrollable={isConnected}
-      >
-        {isConnected ? (
-          activeCard.component
-        ) : (
-          <Fragment>
-            <AccessGate
-              mt={'40px'}
-              Icon={Cavatica}
-              title="Collaborative Analysis"
-              detail="To analyze Kids First data on the cloud, connect to Cavatica."
-            >
-              <ConnectButton
-                onClick={() => {
-                  analyticsTrigger({
-                    property: 'portal',
-                    type: 'recording',
-                    uiArea: TRACKING_EVENTS.categories.user.profile,
-                    action: TRACKING_EVENTS.actions.integration.init,
-                    label: TRACKING_EVENTS.labels.cavatica,
-                  });
-                  effects.setModal({
-                    title: 'How to Connect to Cavatica',
-                    component: (
-                      <CavaticaConnectModal
-                        onComplete={effects.unsetModal}
-                        onCancel={effects.unsetModal}
-                      />
-                    ),
-                  });
-                }}
-              />
-            </AccessGate>
-            <Info
-              link={{
-                url: 'https://kidsfirstdrc.org/support/analyze-data/',
-                text: 'CAVATICA compute cloud platform',
-              }}
-            />
-          </Fragment>
-        )}
-      </DualPaneCard>
-    );
-  },
-);
+const CavaticaProjects = compose(injectState)(({ state: { integrationTokens } }) => {
+  const isConnected = isValidKey(integrationTokens[CAVATICA]);
+
+  return (
+    <DashboardMulticard inactive={!isConnected} tabMenu={['Projects', 'Create']} scrollable>
+      {isConnected ? (
+        [
+          <CavaticaProvider>
+            {({ projects, loading }) => <Connected projects={[]} loading={loading} />}
+          </CavaticaProvider>,
+          <Create />,
+        ]
+      ) : (
+        <NotConnected />
+      )}
+    </DashboardMulticard>
```

Comment:
what do you think about designing the card so it can be consumed like so:
```
({ isConnected }) => {
  const isConnected = isValidKey(integrationTokens[CAVATICA]);

  const onConnect = cardState => someData => {
    cardState.doStuff(someData);
  };
  const onCavaticaData = cardState => someData => {
    cardState.doStuff(someData);
  };
  const onProjectCreationComplete = cardState => someData => {
    cardState.doStuff(someData);
  };
  const onProjectCreationCanceled = cardState => someData => {
    cardState.doStuff(someData);
  };

  return (
    <DashboardMulticard
      tabs={
        isConnected
          ? [
              {
                title: 'Cavatica Projects',
                component: cardState => (
                  <CavaticaProvider onData={onCavaticaData(cardState)}>
                    {stuff => <Connected stuff={stuff} />}
                  </CavaticaProvider>
                ),
              },
              {
                title: 'Create Project',
                component: cardState => (
                  <Create
                    onProjectCreated={onProjectCreationComplete(cardState)}
                    onProjectCreationCanceled={onProjectCreationCanceled(
                      cardState,
                    )}
                  />
                ),
              },
            ]
          : [
              {
                title: 'Oops',
                component: cardState => (
                  <NotConnected onConnect={onConnect(cardState)} />
                ),
              },
            ]
      }
    />
  );
};
```